### DISCUSSION TITLE
About on demand

### DISCUSSION THREAD

**Discussion by fyrus7** [rlogiacco/BatterySense#38] [Q&A] [answered]
Hi, just need a little info. Example sketch comment on demand on pin 3.

Does this mean the metering is updating only after pin 3 is pressed?

Does it only use pin 3 or any pin should update the sensing? Does it need to be spesific pin to function?

**Comment by rlogiacco** [comment, marked as answer]
The pin you specify as `on-demand` is toggled to enable the sensing circuit so that the sensing circuit is activated only when necessary.
The operation is inverted respect to your description: the library pilots the pin when you request the sensing, not the other way around.
You can specify any pin with digital capabilities you specify via the `onDemand` function by calling ti once during the setup/initialization phase: any subsequent call to `level` or `voltage` functions will use the values set by onDemand to perform the sensing.

See:
* https://github.com/rlogiacco/BatterySense#higher-than-5v-activated-on-demand
* http://tinyurl.com/ydbjfk67
* https://github.com/rlogiacco/BatterySense/blob/master/examples/OnDemandDoubleLiIon/OnDemandDoubleLiIon.ino
